hazelcast/src/main/java/com/hazelcast/internal/memory/impl/DirectMemoryBits.java

```diff
@@ -393,53 +553,79 @@ public static char readCharVolatile(long address, boolean bigEndian) {
     }
 
     public static char readCharVolatile(byte[] buffer, int pos, boolean bigEndian) {
-        return readCharVolatile(buffer, (long) (BYTE_ARRAY_BASE_OFFSET + pos), bigEndian);
+        return readCharVolatile(BYTE_BUFFER_MEMORY_ACCESSOR, buffer, pos, bigEndian);
```

Comment:
Why ?
    public static char readCharVolatileB(Object base, long offset) {
        int byte1 = UNSAFE.getByte(base, offset) & 0xFF;
        int byte0 = UNSAFE.getByte(base, offset + 1) & 0xFF;
        return (char) ((byte1 << 8) + byte0);
    }

    public static char readCharVolatileL(Object base, long offset) {
        int byte1 = UNSAFE.getByte(base, offset) & 0xFF;
        int byte0 = UNSAFE.getByte(base, offset + 1) & 0xFF;
        return (char) ((byte0 << 8) + byte1);
    }
I don't see here usage of some readXYZVolatile in current codeBase